zebra/main.c

```diff
@@ -350,6 +359,15 @@ int main(int argc, char **argv)
 		case 'r':
 			retain_mode = 1;
 			break;
+		case 'K':
+			if (keep_kernel_mode) {
+				zlog_err("Keep Kernel mode specified, graceful restart incompatible");
+				return 1;
+			}
+			graceful_restart = atoi(optarg);
+			if (graceful_restart > 5 * GRACEFUL_RESTART_TIME)
```

Comment:
What is the rational behind restricting the max graceful_restart time to 300 seconds ? Even if we want to keep this restriction, I think it is better to throw a CLI error and let user know so that user is aware of this restriction instead of silently restricting this to 300 seconds. or Did i misunderstand this code ?

Reply:
no you didn't.  I was following the lead of the PR I pulled this from and only superficially thought this through when I did so.  I will remove the limitation.